src/audio/kpb.c

```diff
@@ -940,6 +952,7 @@ static void kpb_init_draining(struct comp_dev *dev, struct kpb_client *cli)
 		 * in the history buffer. All we have to do now is to calculate
 		 * read pointer from which we will start draining.
 		 */
+		spin_lock_irq(&root->lock, flags);
```

Comment:
@mrajwa This is lock for interrupt controller. Using it will lock the entire interrupt handling for other cores.

@lgirdwood Why do we even need spinlock here? `irq_local_disable` should be enough as I don't think multiple cores will be ever using the same KPB component instance.

Reply:
Yes I think I should add a custom lock to KPB